processor/src/main/java/org/mapstruct/ap/internal/model/PropertyMapping.java

```diff
@@ -638,16 +639,33 @@ private Assignment forgeMapping(SourceRHS sourceRHS) {
             if ( sourceType.isPrimitive() || targetType.isPrimitive() ) {
                 return null;
             }
+
             String name = getName( sourceType, targetType );
-            ForgedMethod forgedMethod = new ForgedMethod(
-                name,
-                sourceType,
-                targetType,
-                method.getMapperConfiguration(),
-                method.getExecutable(),
-                method.getContextParameters(),
-                getForgedMethodHistory( sourceRHS )
-            );
+            List<Parameter> parameters = new ArrayList( method.getContextParameters() );
+            ForgedMethod forgedMethod;
+            if ( method.isUpdateMethod() ) {
+                parameters.add( Parameter.forForgedMappingTarget( targetType ) );
```

Comment:
I agree with you. I forgot the `getReturnType`.

However, I will still reduce the amount of code. By either new variable which will be either the `targetType` or `TypeKind.VOID` or with a ternary on the same line 😄